azurerm/resource_arm_automation_dsc_nodeconfiguration_test.go

```diff
@@ -0,0 +1,160 @@
+package azurerm
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func TestAccAzureRMAutomationDscNodeConfiguration_basic(t *testing.T) {
+	resourceName := "azurerm_automation_dsc_nodeconfiguration.test"
+	ri := acctest.RandInt()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMAutomationDscNodeConfigurationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMAutomationDscNodeConfiguration_basic(ri, testLocation()),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMAutomationDscNodeConfigurationExists(resourceName),
```

Comment:
Again we should be checking the state of the resource not just that it exists.

Reply:
Added resource attribute check for configuration_name